### PR TITLE
Added scrollable calendar view to home page

### PR DESCRIPTION
**Pull request by Mrez95**
**Design:** 

screen shot 2017-07-12 at 4 52 14 pm

**Changes:**
- Updated homepage to scroll view
- New custom ListView to support UI [bug](https://stackoverflow.com/questions/18367522/android-list-view-inside-a-scroll-view)
- Refactored presenter & fragment to display calendar items in real time and on app load

### CODE REVIEW

File: app/src/main/java/com/example/peterchu/watplanner/home/HomeFragment.java [hunk 1]
```diff
@@ -136,6 +146,7 @@ public void addCourses(List<Course> courses) {
     public void removeCourse(Course course) {
         courseAdapter.remove(course);
         courseAdapter.notifyDataSetChanged();
+        weekView.notifyDatasetChanged();
```
Comment (yongla) [inline app/src/main/java/com/example/peterchu/watplanner/home/HomeFragment.java:174]: It looks like you're telling the week view to refresh its data here to reflect modified data, but the line that actually modifies the week view's data is after this method is called in `onCourseRemoved`. Does this behave properly when testing?

**Review by yongla — COMMENTED** [review]
Left one comment for now. Also: What's the purpose of `ResizableListView`? Why can't the `WeekView` just be directly added to the home screen's layout?

**Comment by yongla** [comment]
Yo, should we show the calendar at the top always? So it doesn't get buried by course cards

File: app/src/main/java/com/example/peterchu/watplanner/home/HomePresenter.java [hunk 2]
```diff
@@ -44,20 +39,34 @@ public void onDataSyncFailure() {
...
+            homeFragment.setCourseSchedule(new ArrayList<CourseComponent>());
+        }
+        for (Course course : courses) {
+            dataRepository.getCourseSchedule( course,
+                new DataRepository.CourseScheduleCallback() {
+                    @Override
+                    public void onCourseScheduleRetrieved(List<CourseSchedule> schedules) {
+                        List<CourseComponent> result = new ArrayList<>();
+                        for (CourseSchedule s : schedules) {
+                            result.addAll(DatabaseHandler.makeCourseComponents(s));
```
Comment (timothytong) [inline app/src/main/java/com/example/peterchu/watplanner/home/HomePresenter.java]: Respect the architecture and dont call anything directly in `DatabaseHandler`

**Comment by timothytong** [comment]
We should investigate why some of the events on the calendar have no text

**Comment by Mrez95** [comment]
@yongla the custom class is introduced as effect of this notorious Android [bug](https://stackoverflow.com/questions/18367522/android-list-view-inside-a-scroll-view). ScrollView and ListView does not work well together, as such, our cards are not being populated dynamically. This solves that.

**Comment by Mrez95** [comment]
@timothytong yes, noticed the same thing for detailed view - let's look at the backend and parsing logic - but that shouldn't block this pr as this is client stuff and just pulling from the backend.

**Review by PeterChu2 — APPROVED** [review]
lgtm! 👍 💯

**Pull request merged**